### PR TITLE
Token unwrap improvements

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

Improve token unwrap so that we can specify the receiver. If no receiver is specified, the "first chain" (i.e.: osmosis) is assumed. 

Still to do:

 - [x] Allow the receiver to be the token's native chain

## Brief Changelog

 * Improvements to the unwrap message
 * Improvements to the tests to be able to test with different receivers

## Testing and Verifying

Added new tests. Existing tests pass

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: cosmwasm/contracts/crosschain-registry/src/execute.rs [hunk 1]
```diff
@@ -811,10 +796,7 @@ mod tests {
         assert_eq!(result.unwrap_err(), expected_error);
         assert_eq!(
             CHANNEL_ON_CHAIN_CHAIN_MAP
-                .load(
-                    &deps.storage,
-                    (&"channel-0".to_string(), &"osmosis".to_string())
-                )
+                .load(&deps.storage, ("channel-0", "osmosis"))
```
Comment (czarcas7ic) [inline cosmwasm/contracts/crosschain-registry/src/execute.rs:799]: Lol, sorry about all these 😅 still getting used to rust types

File: cosmwasm/contracts/crosschain-registry/src/exports.rs [hunk 2]
```diff
@@ -171,8 +180,39 @@ impl<'a> Registries<'a> {
...
+                self.deps.api.debug(&format!("Got error: {e}"));
+                RegistryError::Bech32PrefixDoesNotExist {
+                    chain: chain.into(),
+                }
+            })?;
+        if prefix.is_empty() {
+            return Err(RegistryError::Bech32PrefixDoesNotExist {
+                chain: chain.into(),
+            });
+        }
```
Comment (czarcas7ic) [inline cosmwasm/contracts/crosschain-registry/src/exports.rs:208]: Would this even be possible to store a map from a value to nothing? Or is this just a defense in depth?

File: cosmwasm/contracts/crosschain-registry/src/exports.rs [hunk 3]
```diff
@@ -233,43 +276,30 @@ impl<'a> Registries<'a> {
         Ok(hops)
     }
 
+    /// Returns an IBC MsgTransfer that with a packet forward middleware memo that will send the
+    /// coin back to its original chain and then to the receiver in `into_chain`.
+    /// If the receiver `into_chain` is not specified, the chain this function is called on is assumed.
```
Comment (czarcas7ic) [inline cosmwasm/contracts/crosschain-registry/src/exports.rs:281]: Can we be more specific on what the assumption is? I believe the assumption is the "into_chain" is the chain in which the denom is currently existing that we are unwrapping from.

**Review by czarcas7ic — APPROVED** [review]

File: cosmwasm/contracts/crosschain-registry/src/exports.rs (hunk 3 above)
Comment (nicolaslara) [reply cosmwasm/contracts/crosschain-registry/src/exports.rs:281]: yeah, that's correct. I'll update the comment

File: cosmwasm/contracts/crosschain-registry/src/exports.rs (hunk 2 above)
Comment (nicolaslara) [reply cosmwasm/contracts/crosschain-registry/src/exports.rs:208]: right, I'm not sure if it's possible, haven't tested it, but an empty string is still a string, so probably?

**Pull request merged**